pixeldata/src/lib.rs

```diff
@@ -0,0 +1,597 @@
+        // "pydicom/emri_small_jpeg_ls_lossless.dcm",
+        // "pydicom/gdcm-US-ALOKA-16.dcm",
+        // "pydicom/gdcm-US-ALOKA-16_big.dcm",
+        // "pydicom/image_dfl.dcm",
+        // "pydicom/liver.dcm",
+        // "pydicom/liver_1frame.dcm",
+        // "pydicom/liver_expb.dcm",
+        // "pydicom/liver_expb_1frame.dcm",
+        // "pydicom/meta_missing_tsyntax.dcm",
+        // "pydicom/mlut_18.dcm",
+        // "pydicom/nested_priv_SQ.dcm",
+        // "pydicom/no_meta.dcm",
+        // "pydicom/no_meta_group_length.dcm",
+        // "pydicom/priv_SQ.dcm",
+        // "pydicom/reportsi.dcm",
+        // "pydicom/reportsi_with_empty_number_tags.dcm",
+        // "pydicom/rtdose.dcm",
+        // "pydicom/rtdose_1frame.dcm",
+        // "pydicom/rtdose_expb.dcm",
+        // "pydicom/rtdose_expb_1frame.dcm",
+        // "pydicom/rtdose_rle.dcm",
+        // "pydicom/rtdose_rle_1frame.dcm",
+        // "pydicom/rtplan.dcm",
+        // "pydicom/rtplan_truncated.dcm",
+        // "pydicom/rtstruct.dcm",
+        // "pydicom/test-SR.dcm",
+        // "pydicom/vlut_04.dcm",
+    ])]
+    fn test_parse_dicom_pixel_data(value: &str) {
+        let test_file = dicom_test_files::path(value).unwrap();
+        println!("Parsing pixel data for {:?}", test_file);
+        let obj = open_file(test_file).unwrap();
+        let image = obj.decode_pixel_data().unwrap().to_dynamic_image().unwrap();
+        image
+            .save(format!(
+                "../target/dicom_test_files/pydicom/{}.png",
+                Path::new(value).file_stem().unwrap().to_str().unwrap()
+            ))
+            .unwrap();
+    }
+
+    #[test]
+    fn test_to_ndarray_signed_word() {
+        let test_file = dicom_test_files::path("pydicom/JPEG2000.dcm").unwrap();
+        let obj = open_file(test_file).unwrap();
+        let ndarray = obj
+            .decode_pixel_data()
+            .unwrap()
+            .to_ndarray::<i16>()
+            .unwrap();
+        assert_eq!(ndarray.shape(), &[1024, 256, 1]);
+        assert_eq!(ndarray.len(), 262144);
+        assert_eq!(ndarray[[260, 0, 0]], -3);
+    }
+
+    #[test]
+    fn test_to_ndarray_rgb() {
+        let test_file = dicom_test_files::path("pydicom/SC_rgb_16bit.dcm").unwrap();
+        let obj = open_file(test_file).unwrap();
+        let ndarray = obj
+            .decode_pixel_data()
+            .unwrap()
+            .to_ndarray::<u16>()
+            .unwrap();
+        assert_eq!(ndarray.shape(), &[100, 100, 3]);
+        assert_eq!(ndarray.len(), 30000);
+        assert_eq!(ndarray[[50, 80, 1]], 32896);
+    }
+
+    #[test]
+    fn test_to_ndarray_error() {
+        let test_file = dicom_test_files::path("pydicom/JPEG2000.dcm").unwrap();
+        let obj = open_file(test_file).unwrap();
+        let ndarray = obj.decode_pixel_data().unwrap().to_ndarray::<u8>();
+        if let Err(_) = ndarray {
+            //
+        } else {
+            panic!("should fail");
+        }
```

Comment:
A combination of `assert!` and `matches!` (or `assert_matches!` through a third-party dependency) would make the assertion look better and enable us to be more specific about what kind of error is expected.

Reply:
Ah thanks!